libopenage/util/file.cpp

```diff
@@ -121,5 +121,13 @@ std::ostream &operator<<(std::ostream &stream, const File &file) {
 	return stream;
 }
 
+static File get_temp_file() {
+	std::FILE* tmp_file = std::tmpfile();
+	int temp_fd = fileno(tmp_file);
+	std::string tf_path = "/proc/self/fd/" + std::to_string(temp_fd);
```

Comment:
Wouldn't this solution only work on Linux?

Reply:
I think a better compromise is to get a temporary directory from `get_temp_directory()` and use `std::tmpnam(nullptr)` to get a filename to place into that. Then we don't have to do a cross-plattform "finding out the filename" solution (yet).

Reply:
I agree, this would be a good compromise.